### ISSUE TITLE
Found NaN/Inf in depth0/depth1 gradients!

### ISSUE THREAD

**Issue by russwest44-0** [nianticlabs/mickey#23]
Hi, thanks for the great work!

Epoch 0:  50%|████████████████████████████████████████████████████████████▌                                                            | 719/1438 [27:39<27:39,  2.31s/it, v_num=8]
Found NaN/Inf in depth0/depth1 gradients!████████████████████████████████████████████████████████████████████████████████████████████████████████| 100/100 [03:03<00:00,  1.83s/it]

I'm getting “Found NaN/Inf in depth0/depth1 gradients!” when trying to train the code. What is the problem? Is it because the network is not trained for depth estimation.

**Comment by axelBarroso** [comment]
Hello! Thanks for your interest in our work. 

That is a warning log. Since MicKey training does manual optimization, ie, gradient computation and backpropagation, it checks whether the gradients are valid. If gradients are valid, it will use them to do a backward pass, if they are not (eg, there are NaN values), it will skip them. 

The NaN values come from the pose solver, Procrustes. Procrustes is not very stable when we use a small minimal set as in the training. Increasing the minimal set to 20 correspondences will make it much more stable. However, having a small minimal set mimics the configuration we use during test time. 

Thus, it is expected to see those values, and the training takes care of them. They are only informative. 

Thank you very much, let me know if you have any other questions!
